grakn-graph/src/main/java/ai/grakn/graph/internal/CastingImpl.java

```diff
@@ -65,9 +65,9 @@ public RoleType getRole() {
      * @return The {@link Instance} which is the roleplayer in this casting
      */
     public InstanceImpl getRolePlayer() {
-        Concept concept = getOutgoingNeighbour(Schema.EdgeLabel.ROLE_PLAYER);
-        if(concept != null) {
-            return (InstanceImpl) concept;
+        Optional<Concept> concept = getOutgoingNeighbours(Schema.EdgeLabel.ROLE_PLAYER).findFirst();
+        if(concept.isPresent()) {
+            return (InstanceImpl) concept.get();
         } else {
             return null;
```

Comment:
This will be a fun PR for me!
You can probably simplify this with `orElse`:
```
return concept.orElse(null);
```
Also you can use the generic parameter on `getOutgoingNeighbours` to avoid the cast.

Although obviously I would personally suggest returning an `Optional` :P